### PR TITLE
Add velon supply event

### PR DESCRIPTION
**Pull request by eric6311**
- add Supply event
- add test for supply event

### CODE REVIEW

File: contracts/veLON.sol [hunk 1]
```diff
@@ -242,6 +242,10 @@ contract veLON is IveLON, ERC721, Ownable, ReentrancyGuard {
         }
 
         emit Deposit(msg.sender, _tokenId, _value, _lockedBalance.end, _depositType, block.timestamp);
+        if (_depositType != DepositType.MERGE_TYPE && _depositType != DepositType.INCREASE_UNLOCK_TIME) {
```
Comment (zeqing-guo) [inline contracts/veLON.sol:245]: Line 240 has checked the `_value != 0 && _depositType != DepositType.MERGE_TYPE`. Maybe we could merge these two `if-clause`.

Comment (eric6311) [reply contracts/veLON.sol:245]: I originally thought to put the emit in line 240 is a better way, however the test just couldn't pass. I'm still figure out what was going on.

File: contracts/test/veLON/Deposit.t.sol [hunk 2]
```diff
@@ -34,11 +34,12 @@ contract TestVeLONDeposit is TestVeLON {
         uint256 tokenSupplyBefore = veLon.totalSupply();
         uint256 expectedUnlockTime = block.timestamp + DEFAULT_LOCK_TIME;
 
-        vm.expectEmit(true, true, true, true);
-        emit Deposit(user, tokenSupplyBefore + 1, DEFAULT_STAKE_AMOUNT, expectedUnlockTime, DepositType.CREATE_LOCK_TYPE, block.timestamp);
         uint256 supply = DEFAULT_STAKE_AMOUNT;
         vm.expectEmit(true, true, true, true);
```
Comment (eric6311) [inline contracts/test/veLON/Deposit.t.sol:38]: Make the `Supply` assertion ahead of `Deposit`

File: contracts/veLON.sol [hunk 3]
```diff
@@ -238,6 +238,8 @@ contract veLON is IveLON, ERC721, Ownable, ReentrancyGuard {
         _updateLockedPoint(_tokenId, _oldLocked, _lockedBalance);
 
         if (_value != 0 && _depositType != DepositType.MERGE_TYPE) {
+            tokenSupply = tokenSupply.add(_value);
```
Comment (charlesjhongc) [inline contracts/veLON.sol:241]: Use `prevSupply` to avoid 2nd storage read.

Comment (eric6311) [reply contracts/veLON.sol]: replaced in [196b3c1](https://github.com/consenlabs/tokenlon-contracts/pull/75/commits/196b3c15fb149b918171fda65eefdfeb76da22bd)

**Review by charlesjhongc — APPROVED** [review]

**Pull request merged**